src/Fixer/LanguageConstruct/SingleSpaceAroundConstructFixer.php

```diff
@@ -0,0 +1,462 @@
+
+echo  "Hello!";
+',
+                    [
+                        'constructs_followed_by_a_single_space' => [
+                            'echo',
+                        ],
+                    ]
+                ),
+                new CodeSample(
+                    '<?php
+
+yield  from  baz();
+',
+                    [
+                        'constructs_followed_by_a_single_space' => [
+                            'yield_from',
+                        ],
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before BracesFixer, FunctionDeclarationFixer.
+     * Must run after ModernizeStrposFixer.
+     */
+    public function getPriority(): int
+    {
+        return 36;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        $tokenKinds = array_merge(
+            array_values($this->fixTokenMapContainASingleSpace),
+            array_values($this->fixTokenMapPrecededByASingleSpace),
+            array_values($this->fixTokenMapFollowedByASingleSpace),
+        );
+
+        return $tokens->isAnyTokenKindsFound($tokenKinds) && !$tokens->hasAlternativeSyntax();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $tokenKindsContainASingleSpace = array_values($this->fixTokenMapContainASingleSpace);
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if ($tokens[$index]->isGivenKind($tokenKindsContainASingleSpace)) {
+                $token = $tokens[$index];
+
+                if (
+                    $token->isGivenKind(T_YIELD_FROM)
+                    && 'yield from' !== strtolower($token->getContent())
+                ) {
+                    $tokens[$index] = new Token([T_YIELD_FROM, Preg::replace(
+                        '/\s+/',
+                        ' ',
+                        $token->getContent()
+                    )]);
+                }
+            }
+        }
+
+        $tokenKindsPrecededByASingleSpace = array_values($this->fixTokenMapPrecededByASingleSpace);
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if ($tokens[$index]->isGivenKind($tokenKindsPrecededByASingleSpace)) {
+                $tokens->ensureWhitespaceAtIndex($index - 1, 1, ' ');
+            }
+        }
+
+        $tokenKindsFollowedByASingleSpace = array_values($this->fixTokenMapFollowedByASingleSpace);
+
+        for ($index = $tokens->count() - 2; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind($tokenKindsFollowedByASingleSpace)) {
+                continue;
+            }
+
+            $whitespaceTokenIndex = $index + 1;
+
+            if ($tokens[$whitespaceTokenIndex]->equalsAny([',', ';', ')', [CT::T_ARRAY_SQUARE_BRACE_CLOSE], [CT::T_DESTRUCTURING_SQUARE_BRACE_CLOSE]])) {
```

Comment:
Can we extract all these conditions that do `continue` to separate method and call it here for better readability?

Reply:
copy-pasted from `SingleSpaceAfterConstructFixer`. goal of this PR is not refactoring of existing logic. If you up for it, please raise a (sub?) PR